### ISSUE TITLE
Gutenberg UI updates

### ISSUE THREAD

**Issue by noplanman** [DiasPHPora/wp-to-diaspora#160]
Using the new Gutenberg editor, metabox fields aren't updated correctly after posting.

**Comment by noplanman** [comment]
Also, double-posting may occur due to the way Gutenberg handles post saving.
https://wordpress.org/support/topic/wp-to-diaspora-post-twice/

**Comment by HDValentin** [comment]
I have the same issue. The posts are made twice or more on Diaspora.
I have to delete them by hand.
https://wordpress.org/support/topic/wp-to-diaspora-post-twice/

Blog: https://chillr.de
Diaspora: https://diasp.eu/people/421009ee6cd4a213

**Comment by noplanman** [comment]
@HDValentin I've just released version 3.0.0.

Please give it a go and give me any feedback, thanks!

**Comment by HDValentin** [comment]
I've installed the new Version.
But the issue is already there. 
I posted a new blogpost and on diaspora it was posted twice.

**Comment by noplanman** [comment]
@HDValentin Hmm, could you share what pod you're on and what other plugins you have installed in WordPress?

**Comment by HDValentin** [comment]
The Pod I wrote here: https://github.com/DiasPHPora/wp-to-diaspora/issues/160#issuecomment-575531158

Antispam Bee
Autoptimize
BackWPup
Broken Link Checker
Clean My Archives
Easy Table of Contents
Jetpack by WordPress.com
Limit Login Attempts Reloaded
Mathilda
MC4WP: Mailchimp for WordPress
Plugin Report
Redirection
Relevanssi
Smush
Statify
WordPress Popular Posts
WP-Optimize - Clean. Compress. Cache
WP DoNotTrack
WP to diaspora*
WP YouTube Lyte
Yoast SEO

**Comment by noplanman** [comment]
@HDValentin I think the problem is a race condition of two processes being executed at similar times, where sometimes one manages to finish properly (and therefore the second doesn't need to) or both get processed.

Could you try to edit the file `wp-content/plugins/wp-to-diaspora/lib/class-wp2d-post.php` and add this before line 216:
```php
...

if ( defined( 'REST_REQUEST' ) && REST_REQUEST ) {
	return false;
}

$this->assign_wp_post( $post ); // this should be line 216, to leave as is.

...
```

If this fixes the problem, I'd do some more testing and then release a fix 👍

**Comment by noplanman** [comment]
@HDValentin Have you had a moment to try the code fix proposed above?

**Comment by HDValentin** [comment]
Sorry not jet. Perhaps I can test it this week.

**Comment by HDValentin** [comment]
Yes the code worked.
I just published a blogpost and it was only posted once.
https://diasp.eu/people/421009ee6cd4a213

**Comment by noplanman** [comment]
@HDValentin Great, thanks for the feedback!

I'll release a fixed version in the coming days, hope that solves it for good this time 🤞

**Comment by noplanman** [comment]
Hi @HDValentin, I've created a PR which should be a better fix to the problem.

I'd be super grateful if you could try it, instead of the `REST_REQUEST` code I posted before.

You can find the updated code (which is just moving a few lines further up) here:
https://github.com/DiasPHPora/wp-to-diaspora/pull/171/files#diff-d6e74f775cc5af0051c8dc66fa751dad

Thanks 🙏

**Comment by HDValentin** [comment]
Hi @noplanman, thank you for the optimization.
I am going to try the fix. But I have to write a blog post first. This will need some days.

**Comment by noplanman** [comment]
@HDValentin just a quick ping to check if it's working now, so that I can release the fix 👌

**Comment by HDValentin** [comment]
@noplanman yes it works. I just released a blogpost and it appeared corrext at diaspora

**Comment by noplanman** [comment]
Hi @HDValentin and thanks for your feedback.

I've just released 3.0.2 which contains the fix, hopefully for good now!
